### ISSUE TITLE
How can we pass own certificate for decode, the signed assertion.

### ISSUE THREAD

**Issue by karun-dhir** [fangli/django-saml2-auth#48]
I just want to know where to pass the .cert file or public key in your plugin.

**Comment by nqthqn** [comment]
Does this plugin support signed SAML responses? What does this line mean?

https://github.com/fangli/django-saml2-auth/blob/master/django_saml2_auth/views.py#L107

**Comment by ambsw-technology** [comment]
When a user goes to the `signin` endpoint, they should be redirected to a SAML IdP with a payload.  This option indicates that the payload should not be signed.

Per the underlying [`pysaml2` docs](https://pysaml2.readthedocs.io/en/latest/howto/config.html), a keyfile is provided by including a `key_file` (private) and `cert_file` (public) entry at the top level of the config.  Since `django_saml2_auth` doesn't offer a way to populate this entry, there's no way to provide a file for signature.

**Comment by agger-magenta** [comment]
It should be possible, however, to clone the source code and modify it to accept a `key_file` and a `cert_file`. I'm integrating to an IdP (not run by my own organization) which seems to require a certificate. Anyone who has experience with this?

**Comment by ghost** [comment]
Can the signed assertion mentioned in this thread cause [this](https://github.com/IdentityPython/pysaml2/issues/598)? This module require that the response is signed, but don't give the option to set the certificates to sign, wtf.

**Comment by claytondaley** [comment]
Yes.  This module is "opinionated" insofar as it only supports the simplest use cases.  I [refactored the project](https://github.com/ambsw/django-saml2-auth) to support modification using custom plugins (for example, [custom metadata](https://github.com/ambsw/django-saml2-auth-metadata-inline) and [signout redirection](https://github.com/ambsw/django-saml2-auth-signout-redirect)).  This makes it easier to add functionality without having to incorporate the code into the base project.
